### PR TITLE
Fix issue 34

### PR DESCRIPTION
**Pull request by jeremyweber-np**
Added a property to enable/disable gcr auth checking.  Enabled by default.

### CODE REVIEW

File: plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java [hunk 1]
```diff
@@ -192,6 +192,12 @@ public String getFileName() {
   private JarArchiver jarArchiver;
 
   /**
+   * Allows enabling/disabling gcr credentials
+   */
+  @Parameter(defaultValue = "true", property = "dockerfile.googleregistry")
```
Comment (freesoft) [inline plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java]: If dockerfile.googleregistry property doesn't exist, shouldn't this value have false as default? If that's the case you don't even need to specify defaultValue in the Parameter annotations.

File: plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java [hunk 2]
```diff
@@ -401,13 +407,15 @@ private RegistryAuthSupplier createRegistryAuthSupplier() {
...
     suppliers.add(new ConfigFileRegistryAuthSupplier());
 
-    try {
-      final RegistryAuthSupplier googleSupplier = googleContainerRegistryAuthSupplier();
-      if (googleSupplier != null) {
-        suppliers.add(0, googleSupplier);
-      }
-    } catch (IOException ex) {
-      getLog().info("ignoring exception while loading Google credentials", ex);
+    if ( googleregistry ){
```
Comment (dflemstr) [inline plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java]: Code style

**Comment by dflemstr** [comment]
It would be good to detect the condition that yields GCR unreachable and fail fast. I think having this configuration too is a good thing, but still if people don't discover the config key, this behavior should not happen.

I'm thinking:

  * Setting socket time-outs to something more reasonable
  * Using other ways of detecting a Google-enabled environment, e.g. checking for env vars etc

WDYT @mattnworb ?

**Comment by mattnworb** [comment]
Yes my intent was for this to be as unobtrusive and unnoticeable as possible. I've asked some questions in #34 about this user's environment which would help me to understand the issue more.

File: plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java (hunk 1 above)
Comment (mattnworb) [inline plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java]: I think a more Maven idiomatic name for this parameter and field name would be something like "googleContainerRegistryEnabled". The "docker" prefix is redundant and I think a Boolean flag should end in "Enabled".

Comment (jeremyweber-np) [inline plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java]: @freesoft - my intention was to check for the creds if the property was missing, as that would maintain backward compatibility and it tested out that way.  Let me know if thats not appropriate.

**Comment by jeremyweber-np** [comment]
Ill hold off on addressing anything here consensus is reached in #34.

File: plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java (hunk 1 above)
Comment (mattnworb) [inline plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java]: Looking at this again, I was mistakenly thinking that `property` controlled the name of the `<configuration>` element in pom.xml. For a system property (what `property` actually does, IIRC) then I think the `dockerfile` prefix is ok, although the should maybe be `dockerfile. googleContainerRegistryEnabled`

**Comment by mattnworb** [comment]
Hey @jeremyweber-np regardless of what our investigation in #34 yields I'd be happy to merge this in the meantime if you can address the feedback. I think we ought to have a flag that can turn off this automatic check of having GCR available. Hopefully in #34 we can figure out the cause of that specific issue but this killswitch is good to have anyway as a last resort.

**Comment by mattnworb** [comment]
@jeremyweber-np I ended up addressing my feedback in a new PR (#43) to be able to get this flag out to users. Thanks again for the original contribution!

**Pull request closed without merging**